src/types.js

```diff
@@ -324,3 +324,24 @@ export type PmConversationData = {|
   recipients: string,
   unread: number,
 |};
+
+export type SharedText = {|
+  type: 'text',
+  sharedText: string,
+|};
+
+export type SharedImage = {|
+  type: 'image',
+  sharedImageUrl: string,
+|};
+
+export type SharedFile = {|
+  type: 'file',
+  sharedFileUrl: string,
+|};
+
+/**
+ * The type of data that can be shared to this app from 3rd party apps - either
+ * plain text or an image or a file.
+ */
+export type SharedData = SharedText | SharedImage | SharedFile;
```

Comment:
```suggestion
/**
 * The data we get when the user "shares" to Zulip from another app.
 */
export type SharedData = SharedText | SharedImage | SharedFile;
```

One thing I want to communicate in this jsdoc is that this is the data we *got* from some act of sharing. The "can be shared" sounds like it's something that might in the future be shared, or might not.

One thing that helps tighten up the jsdoc -- makes it shorter, and perhaps also a bit easier to write -- is that the jsdoc on a type focuses on *a value of the type*. Like how `PmConversationData` just above is a "Summary of a PM conversation", and `Identity` is "An identity belonging to this user in some Zulip org", etc. So no need for "The type of …"; and that perhaps helps with the motivation for "can be", too.

The "either plain text or …" is an implementation detail and so can just be left out. That also avoids a risk of another alternative being added later and the jsdoc not getting updated :wink: